modules/tensor_mechanics/src/actions/DynamicTensorMechanicsAction.C

```diff
@@ -44,6 +44,8 @@ validParams<DynamicTensorMechanicsAction>()
 DynamicTensorMechanicsAction::DynamicTensorMechanicsAction(const InputParameters & params)
   : TensorMechanicsAction(params)
 {
+  if (_use_ad)
+    mooseError("In ", _name, ": AD not setup for use with DynamicTensorMechanicsAction");
```

Comment:
How about making all those errors `paramError("use_ad", "AD not setup for use with DynamicTensorMechanicsAction")` etc.